src/atoms_calculators.jl

```diff
@@ -0,0 +1,76 @@
+#
+# Define AtomsCalculators interface for DFTK.
+#
+# This interface is inspired by the one used in Molly.jl, 
+# see https://github.com/JuliaMolSim/Molly.jl/blob/master/src/types.jl
+#
+# TODO: Decide how Hamiltonian terms should be updated upon change of the positions.
+# These terms are held inside the `basis`. Currently, the whole basis is rebuilt.
+# TODO: Find out if symmetries can be re-enabled (see issue on GH).
+
+using AtomsBase
+using AtomsCalculators
+
+Base.@kwdef struct DFTKParameters
+    model_kwargs = (; )
+    basis_kwargs = (; )
+    scf_kwargs = (; )
+end
+
+struct DFTKState
+    scfres::NamedTuple
+end
+function DFTKState(system::AbstractSystem, params::DFTKParameters)
+    basis = prepare_basis(system, params)
+    ρ = guess_density(basis)
+    ψ = nothing # Will get initialized by SCF.
+    return DFTKState((;ρ, ψ, basis))
+end
+
+function prepare_basis(system::AbstractSystem, params::DFTKParameters)
+    # By default create an LDA model.
+    model = model_LDA(system; symmetries=false, params.model_kwargs...)
```

Comment:
This should be a little more flexible to be useful. Use `model_DFT` here, which takes the XC functionals as an argument after the system, a bit like this:
https://github.com/mfherbst/AiidaDFTK.jl/blob/master/src/AiidaDFTK.jl#L52
These you can also store in DFTKParameters inside the `model_kwargs` maybe and then use a `pop!` or similar to single the `xc` arguments out. Alternatively you can also make an extra method of `model_DFT` that takes the `xc` as a kwarg (that might actually be the better solution here)